### PR TITLE
Fix dry sponge delete waterlogged block & Fix dry sponge can't absorb water at layer 1

### PR DESCRIPTION
**Pull request by GoodLucky777**
Fix dry sponge delete waterlogged block (#643) & Fix dry sponge can't absorb water at layer 1 (#94, #430)

TODO:
  - [x] Fix absorb problem

### CODE REVIEW

**Comment by RizenSoulja** [comment]
Yesh!

**Review by LoboMetalurgico — CHANGES_REQUESTED** [review]
Hello! Thanks for your Pull Request, but for the tests I did, the sponge's behavior remains the same. Could you check this?

File: src/main/java/cn/nukkit/block/BlockSponge.java [hunk 1]
```diff
@@ -112,13 +112,13 @@ private boolean performWaterAbsorb(Block block) {
                     if (faceBlock.getId() == BlockID.BLOCK_KELP || faceBlock.getId() == BlockID.SEAGRASS || faceBlock.getId() == BlockID.SEA_PICKLE || faceBlock instanceof BlockCoralFan) {
                         faceBlock.getLevel().useBreakOn(faceBlock);
                     }
-                    this.getLevel().setBlockStateAt(faceBlock.getFloorX(), faceBlock.getFloorY(), faceBlock.getFloorZ(), 1, Block.get(BlockID.AIR));
+                    this.getLevel().setBlockStateAt(faceBlock.getX(), faceBlock.getY(), faceBlock.getZ(), 1, Block.get(BlockID.AIR));
```
Comment (roridev) [inline src/main/java/cn/nukkit/block/BlockSponge.java]: The whole basis of this pull request relies on the change from `floorX` to `X` affecting the sponge's behaviour.

Why is that the case and why floored coordinates would be a problem?

As @LoboMetalurgico stated there is no reason why the simple change to exact coordinates would change any behaviour.

For this reason, until this change is explained or we're able to notice the fixes at all we can't merge this.

**Review by roridev — DISMISSED** [review]

**Comment by GoodLucky777** [comment]
It was looping and also checking air so I fixed, Ready to be reviewed

**Review by LoboMetalurgico — APPROVED** [review]
Looks like everything is fine now. Thank you!

**Pull request merged**